### PR TITLE
Add output reset API skeleton

### PR DESCRIPTION
**Pull request by gz**
Adds API skeleton/necessary platform support for https://github.com/feldera/feldera/issues/5980.

Makes it easier to test it because the platform changes are available afterwards.

### Describe Incompatible Changes

None, reset will return a 501 not implemented error.

### CODE REVIEW

File: crates/pipeline-manager/src/api/endpoints/pipeline_interaction.rs [hunk 1]
```diff
@@ -460,6 +460,85 @@ pub(crate) async fn get_pipeline_output_connector_status(
     Ok(response)
 }
 
+/// Reset Output Connector
+///
+/// Reset an output connector configured in `snapshot_and_follow` mode.
+///
+/// This clears buffered output, asks the sink to reset itself, and then replays
+/// a full snapshot before resuming incremental updates.
```
Comment (mythical-fred) [inline crates/pipeline-manager/src/api/endpoints/pipeline_interaction.rs:468]: These docs describe the eventual reset semantics as if they exist today (`clears buffered output`, `asks the sink to reset itself`, `replays a full snapshot`). But this PR wires the manager endpoint to a pipeline endpoint that always returns `not_supported`, so the public API contract is misleading right now. Please either document this as a placeholder/skeleton that currently returns not implemented, or wait to publish these semantics until the implementation exists.

File: crates/pipeline-manager/src/api/endpoints/pipeline_interaction.rs [hunk 2]
```diff
@@ -460,6 +460,85 @@ pub(crate) async fn get_pipeline_output_connector_status(
...
+                ("Disconnected during response" = (value = json!(examples::error_pipeline_interaction_disconnected()))),
+                ("Response timeout" = (value = json!(examples::error_pipeline_interaction_timeout())))
+            )
+        ),
+        (status = INTERNAL_SERVER_ERROR, body = ErrorResponse),
+    ),
+    tag = "Output Connectors"
+)]
+#[post("/pipelines/{pipeline_name}/views/{view_name}/connectors/{connector_name}/reset")]
+pub(crate) async fn post_pipeline_output_connector_reset(
```
Comment (mythical-fred) [inline crates/pipeline-manager/src/api/endpoints/pipeline_interaction.rs:504]: This adds a new user-visible API across the manager -> pipeline boundary, but the PR has no unit/integration coverage and no manual test plan. Per the review rules, behavior changes need tests, and cross-boundary behavior usually needs integration coverage. Please add at least a focused request/response test for the new reset endpoint path and status mapping before merging.

**Review by mythical-fred — CHANGES_REQUESTED** [review]
Two blockers, see inline.

**Review by ryzhyk — APPROVED** [review]
Reset may possibly be useful for non-snapshot-and-follow operators as well.
But I think this is ok to experimentation purposes.
